### DISCUSSION TITLE
How can I set the area ratio of LIDs in subcatchments in pyswmm?

### DISCUSSION THREAD

**Discussion by Nanzulo** [pyswmm/pyswmm#460] [Q&A]
I have an .inp file of an urban sewer system that includes green roofs.
In SWMM, I can set different area ratios for green roofs in each subcatchment by using LID usage editor. Ex, 2.7% green roof area is set in subcatchment A.
![image](https://github.com/pyswmm/pyswmm/assets/126145166/243f72a1-fe0c-449e-aa2e-48c704eeae74)

Can I achieve the same in pyswmm?
I have checked the functions related to subcatchment modules and the LidControls module in the pyswmm API. However, there is no specific function to set the area ratio of LIDs in subcatchments. I can only modify for LIDs parameters like soil thickness, drain coefficient, pavement regeneration...

If there is no related functions in pyswmm, could someone suggest an alternative method in python?
I would greatly appreciate any comments or advice that can be provided. Thank you!

**Comment by karosc** [comment]
@Nanzulo , this can be done using [LID Group Units](http://pyswmm.github.io/pyswmm/reference/api/pyswmm.lidgroups.LidUnit.html#pyswmm.lidgroups.LidUnit), which allow you to get/set the number of units and area of each unit **prior to the start of a simulation**

Although the docs don't have examples of this, the [unit tests have an example](https://github.com/pyswmm/pyswmm/blob/22aed805ec435dce93d0ccc084041e380827e01d/pyswmm/tests/test_lid.py#L75C28-L128) you might be able to follow.

**Comment by bemcdonnell** [comment]
@Nanzulo, I prepared a little code block for another user to help them through the LID part of pyswmm.  It might help you too:

[lid_example.zip](https://github.com/pyswmm/pyswmm/files/12017019/lid_example.zip)
